python-stdlib/argparse/argparse.py

```diff
@@ -103,6 +115,12 @@ def add_argument(self, *args, **kwargs):
                 dest = args[0]
             if not args:
                 args = [dest]
+        arg_type = kwargs.get("type", None)
+        if arg_type is not None:
+            if not callable(arg_type):
+                raise ValueError("type is not callable")
+            if default is not None and not isinstance(default, str):
```

Comment:
This is not correct: it skips conversion when an argument is passed. Test like

```
parser.add_argument("-x", type=CustomArgType(1), default=1)
parser.add_argument("-g", type=CustomArgType(1), default=1)
parser.add_argument("-i", type=CustomArgType(1), default="1")
args = parser.parse_args(["-x", "3"])
print(args)
assert args.x == 4
assert args.g == 1
assert args.i == 2
```

Reply:
That was it, missed that bit.  Thanks for sticking with this, it was indeed an issue on my end, code-wise.

I'd partially chalk it up to CPython's documentation being not really clear for that.  It should probably read (emphasis mine) "If the type keyword is used with the default keyword, the type converter is only applied *to the default value* if the default is a string.".

Still, the test file's behaviour in isolation under CPython now matches MicroPython's when using its own `argparse` implementation.  Hope this should be it :)